src/database-controller/src/write-merger/handler.js

```diff
@@ -279,9 +279,45 @@ async function putFrameworkRequest(req, res, next) {
   }
 }
 
+async function archiveFramework(req, res, next) {
+  // The handler to handle framework archive
+  // Only completed job can be archived. If job is not a completed job, return 400 Bad Request.
+  // If the job is not found, return 404 not found.
+  // If job is successfully archived, return 200 OK.
+  // If other error happens, return 500.
+  try {
+    const frameworkName = req.params.frameworkName;
+    if (!frameworkName) {
+      return next(createError(400, 'Cannot find framework name.'));
+    }
+    await lock.acquire(frameworkName, async () => {
+      const framework = await databaseModel.Framework.findOne({
+        attributes: ['subState'],
+        where: { name: frameworkName },
+      });
+      if (!framework) {
+        // if the old framework doesn't exist, report a 404 error
+        throw createError(404, `Cannot find framework ${frameworkName}.`);
+      } else if (framework.subState !== 'Completed') {
+        throw createError(400, `Framework ${frameworkName} is not completed.`);
+      } else {
+        // If the old framework exists and is completed, try to set archived=true
+        await databaseModel.Framework.update(
```

Comment:
The word 'archive' is not accurate, prefer 'hide' as it only stills stores in DB, and occupy the job name.

BTW, if another user submit a job naming conflict with a 'hide' job, the user has no idea, as he cannot get the conflict 'hide' job by default.

So, could we do not introduce this feature?
Intead, I prefer 'archive' to be:
1. Directly delete the job in DB, and move it to other backlog storage or even just do not backup
2. Split the API and webportal view between jobs in DB and jobs in the backlog storage.

@fanyangCS  to more comment on this

Reply:
> > if a job has been archived, does it appear in the history server?
> 
> After DB, there is no history job concept, only job previous retry attempts are stored in history job table.
> For my previous comment "backlog storage" can be the "history job table", but this PR still keeps the archived job in "current job table" which occupy the job namespace.
> 
> > I feel the definition of "archive" might be a bit complicated to end user. How about we simply call it delete? A deleted job won't appear in history server or job list view.
> > And we can keep all the deleted jobs in a separate table or separate DB. User can view their deleted jobs but cannot restore or further delete it (for auditing issue).
> > Admin can delete rows in the table.
> 
> Agree with this. But seems the orignal requirement from user, is that:
> 
> > As an Administrator, I could **able to archive some bad job to clean up the view for all the users**. After archive, the jobs are disappeared from job list and search views, but accessible from the job url (as archived read only job). Admin and user may or may not able to restore this jobs (to be decides, comments are welcomed here).
> 
> Looks like a authz requirement that Admin want to kill and hide some bad jobs, so that only admin can see it again.
> 
> Anyway, we need to figure out the requirement and a detailed design review before this PR.

We can explicitly provide a web page to show the deleted jobs. We need a new API to retrieve deleted jobs.

Reply:
Yep. But this looks like not traditional 'archive' feature (move old ones to backlog to improve serving perf), now I am feeling our goal (and user requirement) is not clear, we would better to first figure out them. As my latest comment, job label may be enough for them.